java/code/src/com/redhat/rhn/manager/action/ActionManager.java

```diff
@@ -1950,6 +1952,41 @@ public static Action schedulePackageAction(User scheduler, List<Map<String, Long
         return action;
     }
 
+
+    private static List<Map<String, ? extends Object>> removeDuplicatedName(List<Map<String, Long>> packageMaps) {
+        Map<String, Map<String, Long>> packageMapsWithoutDuplicated = new HashMap<>();
+
+        for (Map<String, Long> map : packageMaps) {
+
+            Map<String, Long> newMap = map.entrySet().stream()
+                 .collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue()));
+
+            Map<String, Long> previous = packageMapsWithoutDuplicated.put(
+                    newMap.get("name_id").toString(), newMap);
```

Comment:
I think we have no way to influence which package will win, right?
Or it is the last package in the list, when I get it right?

Reply:
yes, currently wins the last package in the list. 
Probably the right thing to do it's install  the one with newer version. 
**Maybe** the last package it's already the newest one (I'm not sure about it), otherwise we can parse and compare the version (in whatever format would be). I didn't implement the version comparison because I was worried to over-engineering the problem (considering that this problem should happens rarely).  What do you think about it?

Reply:
I think this is ok